flink-runtime/src/main/java/org/apache/flink/streaming/api/windowing/assigners/WindowAssigner.java

```diff
@@ -64,6 +66,13 @@ public abstract Collection<W> assignWindows(
      */
     public abstract Trigger<T, W> getDefaultTrigger();
 
+    /** Returns the default async trigger associated with this {@code WindowAssigner}. */
+    @Experimental
+    public AsyncTrigger<T, W> getDefaultAsyncTrigger() {
+        // Subclass should override this method if they support async state processing.
+        throw new RuntimeException("The assigner does not support async state.");
+    }
+
```

Comment:
Seems useless for now? Can we remove this?

Reply:
Resolved. 

Yes, `getDefaultAsyncTrigger()` is no use, and I have removed it from `WindowAssigner`.